src/App.js

```diff
@@ -5,39 +5,73 @@ import { useState, useEffect } from 'react';
 import AddItem from './components/AddItem/AddItem';
 import NavLinks from './components/Navigation/NavLinks';
 import ItemList from './pages/ItemList';
+import { db } from './lib/firebase';
+import { collection, getDocs, query, where } from 'firebase/firestore';
 
 function App() {
-  const [token, setToken] = useState(null);
+  const [tokenPresent, setTokenPresent] = useState(null);
+  const [token, setToken] = useState('');
   const navigate = useNavigate();
+  const [submittedToken, setSubmittedToken] = useState('');
+
+  const handleItemNameChange = (e) => setSubmittedToken(e.target.value);
 
   useEffect(() => {
     const json = localStorage.getItem('shoppingListToken');
     const loadedToken = JSON.parse(json);
     if (loadedToken) {
-      navigate('/item-list');
       setToken(loadedToken);
+      setTokenPresent(true);
```

Comment:
In our office hours yesterday, we discussed that this `tokenPresent` state variable isn't needed as you can derive this state from `!!token`. 
Generally, if you can derive one piece of state from another, then you want to avoid creating the duplicate state.

Reply:
Great point! Will change it before merging.